app/views/compute_resources_vms/form/azurerm/_base.html.erb

```diff
@@ -155,10 +155,10 @@
 %>
 
 <%= selectable_f f, :os_disk_caching, %w(None ReadOnly ReadWrite),
-               {},
+               {   :include_blank => _('Please select OS Disk Caching method') },
                {
                    :label    => _('OS Disk Caching'),
-                   :required => true,
+                   :label_help => _("Default ReadWrite"),
```

Comment:
The current code will not default to `ReadWrite`, [this](https://github.com/theforeman/foreman_azure_rm/blob/d75bcba8188b0508a49252328e57bb6d89bf37de/app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb#L16) line will always evaluate as false. So unless Azure will set it up for us, we will get nil there.

Reply:
I agree as per the line you mentioned that will evaluate to false. But, it still defaults to ReadWrite after applying this patch may be due to this [line](https://github.com/theforeman/foreman_azure_rm/blob/d75bcba8188b0508a49252328e57bb6d89bf37de/app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb#L24-L26).